runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8693,162 +8853,9 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
-            if (compiler->getOption(TR_EnableSelfTuningScratchMemoryUsageBeforeCompile))
-               {
-               bool incompleteInfo = false;
-               // Abort the compile if we don't have at least getScratchSpaceLowerBound()
-               // available, plus some safe reserve
-               // TODO: we may want to use a lower value for third parameter below if the
-               // compilation is deemed cheap (JNI, thunks, cold small method)
-               uint64_t physicalLimit = that->_compInfo.computeFreePhysicalLimitAndAbortCompilationIfLow(compiler,
-                                                                                                   incompleteInfo,
-                                                                                                   TR::Options::getScratchSpaceLowerBound());
-               // If we were able to get the memory information
-               if (physicalLimit != OMRPORT_MEMINFO_NOT_AVAILABLE)
-                  {
-                  // If the proposed scratch space limit is greater than the available
-                  // physical memory, we need to lower the scratch space limit.
-#if defined(J9VM_OPT_JITSERVER)
-                  // Moreover, for JITServer do not allow a single compilation to consume
-                  // more than half of the free physical memory
-                  if (compiler->isOutOfProcessCompilation())
-                     physicalLimit = std::max(physicalLimit >> 1, static_cast<uint64_t>(TR::Options::getScratchSpaceLowerBound()));
-#endif
-                  if (proposedScratchMemoryLimit > physicalLimit)
-                     {
-                     if (incompleteInfo)
-                        {
-                        // If we weren't able to get all the memory information
-                        // only lower the limit for JSR292 compilations,
-                        // but not beyond the default value for scratch memory
-                        if (isJSR292)
-                           {
-                           proposedScratchMemoryLimit = (physicalLimit >= scratchSegmentProvider.allocationLimit()
-                                                         ? physicalLimit
-                                                         : scratchSegmentProvider.allocationLimit());
-                           }
-                        }
-                     else // We have complete memory information
-                        {
-                        proposedScratchMemoryLimit = physicalLimit;
-                        }
-                     }
-                  }
-               }
-
-            // Cap the limit for JIT to 4GB
-            size_t proposedCapped = proposedScratchMemoryLimit > UINT_MAX ? UINT_MAX : (size_t)proposedScratchMemoryLimit;
-            scratchSegmentProvider.setAllocationLimit(proposedCapped);
-            }
+         compiler = compiler_tmp;
```

Comment:
I think the temporary variable `compiler_tmp` is a workaround to the following error during the `make all` step of the build process: `error: binding reference of type ‘TR::Compilation*&’ to ‘TR::Compilation* volatile’ discards qualifiers`.